File: Categories/Array/2352_equalRowAndColumnPairs.cs
public class Solution {
    public int EqualPairs(int[][] grid) {
        int numRows = grid.Length, numCols = grid[0].Length;
        Dictionary<string, int> rowEmbeddingCounter = new Dictionary<string, int>();

        for (int rowIdx = 0; rowIdx < numRows; rowIdx+= 1) {
            string rowEmbedding = String.Join(",", grid[rowIdx]);
            if (!rowEmbeddingCounter.ContainsKey(rowEmbedding)) {
                rowEmbeddingCounter.Add(rowEmbedding, 1);
            }
            else {
                rowEmbeddingCounter[rowEmbedding] += 1;
            }
        }

        int cnt = 0;
        List<int> col = new List<int>();
        for (int colIdx = 0; colIdx < numCols; colIdx += 1) {
            col.Clear();
            for (int rowIdx = 0; rowIdx < numRows; rowIdx += 1) {
                col.Add(grid[rowIdx][colIdx]);
            }
            string colEmbedding = String.Join(",", col);
            cnt += rowEmbeddingCounter.ContainsKey(colEmbedding) ? rowEmbeddingCounter[colEmbedding] : 0;
        }

        return cnt;
    }
}

File: Categories/Array/2352_equalRowAndColumnPairs.py
from typing import *
from collections import defaultdict


class Solution:
    def equalPairs(self, grid: List[List[int]]) -> int:
        num_rows, num_cols = len(grid), len(grid[0])
        row_embedding_cntr = defaultdict(int)
        for row in grid:
            row_embedding_cntr[",".join(map(str, row))] += 1

        cnt = 0
        for col_idx in range(num_cols):
            col = []
            for row_idx in range(num_rows):
                col.append(grid[row_idx][col_idx])
            
            col_embedding = ",".join(map(str, col))
            if col_embedding in row_embedding_cntr:
                cnt += row_embedding_cntr[col_embedding]

        return cnt